catalog/java/io/material/catalog/lists/MaterialListAdapterDemoFragment.java

```diff
@@ -0,0 +1,141 @@
+import android.widget.AutoCompleteTextView;
+
+import androidx.annotation.Nullable;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.google.android.material.lists.MaterialListAdapter;
+import com.google.android.material.lists.MaterialListBinder;
+import com.google.android.material.lists.viewholders.SingleLineItemViewHolder;
+import com.google.android.material.lists.viewholders.ThreeLineItemViewHolder;
+import com.google.android.material.lists.viewholders.TwoLineItemViewHolder;
+
+import java.util.Arrays;
+import java.util.List;
+
+import io.material.catalog.R;
+import io.material.catalog.feature.DemoFragment;
+
+/** A fragment that uses MaterialListAdapters to display different lists of cards */
+public class MaterialListAdapterDemoFragment extends DemoFragment {
+
+  @Override
+  @SuppressWarnings("RestrictTo")
+  public View onCreateDemoView(
+      LayoutInflater layoutInflater, @Nullable ViewGroup viewGroup, @Nullable Bundle bundle) {
+
+    View layout = layoutInflater.inflate(R.layout.cat_material_list_adapter_fragment, viewGroup, false);
+
+    RecyclerView recyclerView = layout.findViewById(R.id.recycler_view);
+    recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
+    recyclerView.setAdapter(materialListAdapters[0]);
+
+    final List<String> options = Arrays.asList("Single Line Item", "Two Line Item", "Three Line Item");
+
+    AutoCompleteTextView editTextFilledExposedDropdown =
+        layout.findViewById(R.id.filled_exposed_dropdown);
+    editTextFilledExposedDropdown.setText(options.get(0));
+    editTextFilledExposedDropdown.setAdapter(new ArrayAdapter<>(
+        getContext(),
+        R.layout.cat_popup_item,
+        options));
+    editTextFilledExposedDropdown.addTextChangedListener(new TextWatcher() {
+      @Override
+      public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+      }
+
+      @Override
+      public void onTextChanged(CharSequence s, int start, int before, int count) {
+      }
+
+      @Override
+      public void afterTextChanged(Editable s) {
+        editTextFilledExposedDropdown.setAdapter(new ArrayAdapter<>(
+            getContext(),
+            R.layout.cat_popup_item,
+            options));
+        recyclerView.setAdapter(materialListAdapters[options.indexOf(s.toString())]);
+      }
+    });
+
+    return layout;
+  }
+
+  //Defining a dataset of twenty Strings
+  String[] listData = new String[]{
+      "First Item",
+      "Second Item",
+      "Third Item",
+      "Fourth Item",
+      "Fifth Item",
+      "Sixth Item",
+      "Seventh Item",
+      "Eighth Item",
+      "Ninth Item",
+      "Tenth Item",
+      "Eleventh Item",
+      "Twelfth Item",
+      "Thirteenth Item",
+      "Fourteenth Item",
+      "Fifteenth Item",
+      "Sixteenth Item",
+      "Seventeenth Item",
+      "Eighteenth Item",
+      "Nineteenth Item",
+      "Twentieth Item"};
```

Comment:
What about converting this to a resource string such that it can be translated in the future? You can then loop the single resource string 20 times and have a `%s` that is replaced with the number.

For e.g.:

```xml
<string name="cat_list_item_prefix">Item %d</string>
```

```java
private List<String> getListData() {
  List<String> listData = new ArrayList();
  for (int x = 1; x <= 20; x++) {
    listData.add(resources.getString(R.string.cat_list_item_prefix, x));
  }
  return listData;
}
```

Reply:
The idea behind the demo is to show how to use Material Lists with a dataset. I define the dataset explicitly to prevent confusion and have readers understand the dataset. This later helps the reader understand the list code, rather than not knowing what the dataset looks like. 

I originally used Collections.nCopies() to automatically create items, but then I found the code harder to read and understand. This code is built for readability and not efficiency.